### PR TITLE
libevent: Make OpenSSL dependency optional

### PR DESCRIPTION
**Pull request by aneeshusa**
###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `build-use-sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [ ] NixOS
   - [ ] macOS
   - [x] Linux
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

**Comment by Mic92** [comment]
Please also change this pull request to staging.

**Comment by fpletz** [comment]
@Mic92 But this change shouldn't change the hash of libevent?

File: pkgs/development/libraries/libevent/default.nix [hunk 1]
```diff
@@ -11,13 +16,20 @@ stdenv.mkDerivation rec {
...
+    ;
   outputBin = "dev";
-  propagatedBuildOutputs = [ "out" "openssl" ];
+  propagatedBuildOutputs = [ "out" ]
+    ++ stdenv.lib.optional enableOpenssl "openssl"
+    ;
 
-  buildInputs = [ openssl ] ++ stdenv.lib.optional stdenv.isCygwin findutils;
+  buildInputs = []
+    ++ stdenv.lib.optional enableOpenssl "openssl"
```
Comment (fpletz) [inline pkgs/development/libraries/libevent/default.nix]: openssl shouldn't be a string here but the reference to the package. :wink:

File: pkgs/development/libraries/libevent/default.nix [hunk 2]
```diff
@@ -1,4 +1,9 @@
-{ stdenv, fetchurl, openssl, findutils }:
+{
+  stdenv, fetchurl, findutils,
```
Comment (rycee) [inline pkgs/development/libraries/libevent/default.nix]: This formatting is a bit unusual. Typically the comma is at the beginning of the line. Something like
```
{ stdenv, fetchurl, findutils
, enableOpenssl ? true, openssl
}

File: pkgs/development/libraries/libevent/default.nix [hunk 3]
```diff
@@ -1,4 +1,9 @@
...
+  enableOpenssl? true, openssl
```
Comment (rycee) [inline pkgs/development/libraries/libevent/default.nix]: It seems that `sslSupport` is more commonly used in Nixpkgs. So I would suggest to use `sslSupport ? true` here.

File: pkgs/development/libraries/libevent/default.nix (hunk 1 above)
Comment (aneeshusa) [reply pkgs/development/libraries/libevent/default.nix]: Oops, you can tell I never actually tried this with `openssl ` enabled...thanks for catching that.

File: pkgs/development/libraries/libevent/default.nix (hunk 2 above)
Comment (aneeshusa) [reply pkgs/development/libraries/libevent/default.nix]: I've seen the comma go both at the beginning and the end and I also find it a bit unbalanced to have a bunch of items on the first line with the open bracket but none on the last line with the close bracket. That being said, it's a minor quibble so I've changed to your recommendation here.

**Pull request merged**